src/qvimagecore.cpp

```diff
@@ -250,13 +259,14 @@ QList<QVImageCore::CompatibleFile> QVImageCore::getCompatibleFiles()
         QString mimeType;
         if (!matched || sortMode == 3)
         {
-            mimeType = mimeDb.mimeTypeForFile(fileInfo).name();
+            mimeType = mimeDb.mimeTypeForFile(absoluteFilePath, mimeMatchMode).name();
```

Comment:
I intentionally passed in `absoluteFilePath` instead of `fileInfo` here to work around a performance issue wherein QMimeDatabase will check isDir() if a QFileInfo is passed in (so it can return "inode/directory" when appropriate), even if the MatchExtension mode was requested.